### ISSUE TITLE
Can't set timezone: "Failed to set time zone: Failed to set time zone: Is a directory"

### ISSUE THREAD

**Issue by salfter** [flatcar/Flatcar#491]
## Description

I have a Home Assistant container that I moved from Gentoo to Flatcar.  It's now firing time-based events 7 hours early because the system clock is set to UTC instead of local time.  I checked the container on the Gentoo box from which I copied it; it appears to just use whatever timezone the host uses (US/Pacific).  I found the [instructions to set the timezone](https://kinvolk.io/docs/flatcar-container-linux/latest/setup/customization/configuring-date-and-timezone/) and issued this:

```sudo timedatectl set-timezone America/Los_Angeles```

Instead of setting the timezone, it responded with this:

> Failed to set time zone: Failed to set time zone: Is a directory

(Since Flatcar is descended from Gentoo, I've worked around the problem for now by copying /etc/localtime from one of my Gentoo boxes.  I suspect, however, that this is a suboptimal solution.)

## Impact

The lamp that's supposed to turn on at sunset is turning on in the middle of the day.

## Environment and steps to reproduce

1. **Set-up**: bare-metal installation created with flatcar-install.sh, running on a spare server for now
2. **Task**: 
3. **Action(s)**: 
  a. log into the Flatcar server
  b. ```sudo timedatectl set-timezone America/Los_Angeles```
4. **Error**: 
> Failed to set time zone: Failed to set time zone: Is a directory

## Expected behavior

system timezone changed from UTC to Pacific time

**Comment by tormath1** [comment]
Hi @salfter, 

What is your current Flatcar's version (`cat /etc/os-release`) ? I tried to reproduce using the current stable (`2905.2.2`) and I can successfully set the timezone using `timedatectl`.

I can reproduce if I `sudo mkdir /etc/localtime` - maybe you created this directory by mistake ? 

For your information, if you use `ignition` for initial provisioning, you can directly create the symlink to set the timezone:
```json
{
  "ignition": { "version": "2.2.0" },
  "storage": {
    "links": [{
      "filesystem": "root",
      "path": "/etc/localtime",
      "overwrite": true,
      "target": "/usr/share/zoneinfo/America/Los_Angeles"
    }]
  }
}
```

**Comment by salfter** [comment]
> What is your current Flatcar's version 

2905.2.2.

> I can reproduce if I sudo mkdir /etc/localtime - maybe you created this directory by mistake ?

The only reference to /etc/localtime in the YAML I'm using is to include it on a read-only basis in the Home Assistant container (which would explain why it shares the same timezone as the host).  I suspect this might've created /etc/localtime as a directory when it didn't already exist, since I hadn't previously set the timezone.

I deleted the localtime file I'd copied from a Gentoo box and tried timedatectl again.  This time, it worked.  It created a symlink to one of the files under /usr/share/zoneinfo.  I took a look inside the Home Assistant container; it has the same zoneinfo files in the same place, so the time's good there as well.  

I've fixed my YAML file according to your suggestion so that the timezone is set with a link instead of a copied file.  I'll be able to test it out later (need to rebuild the system anyway as I had to update most of the *.service files to allow containers to persist, but that's another matter).  I think it's safe to close this out.  Thanks!
